### PR TITLE
Fix game commands being directly executed from within network_update.

### PR DESCRIPTION
**Pull request by ZehM4tt**
Refactored the network code to use the game command queue to not directly execute game commands from within network_update. This should also fix false positives when calling scenario_rand within game command callbacks.

### CODE REVIEW

**Comment by janisozaur** [comment]
Replace tabs with spaces and update network version.

File: src/openrct2/network/network.cpp [hunk 1]
```diff
@@ -2007,7 +2032,7 @@ void Network::Server_Handle_GAMECMD(NetworkConnection& connection, NetworkPacket
             ticks - connection.Player->LastPlaceSceneryTime < ACTION_COOLDOWN_TIME_PLACE_SCENERY &&
             // In case platform_get_ticks() wraps after ~49 days, ignore larger logged times.
             ticks > connection.Player->LastPlaceSceneryTime
-        ) {
+            ) {
```
Comment (janisozaur) [inline src/openrct2/network/network.cpp]: Here

File: src/openrct2/network/network.cpp [hunk 2]
```diff
@@ -2021,38 +2046,20 @@ void Network::Server_Handle_GAMECMD(NetworkConnection& connection, NetworkPacket
             ticks - connection.Player->LastDemolishRideTime < ACTION_COOLDOWN_TIME_DEMOLISH_RIDE &&
             // In case platform_get_ticks() wraps after ~49 days, ignore larger logged times.
             ticks > connection.Player->LastDemolishRideTime
-        ) {
+            ) {
```
Comment (janisozaur) [inline src/openrct2/network/network.cpp]: Here

File: src/openrct2/network/network.cpp [hunk 3]
```diff
@@ -2021,38 +2046,20 @@ void Network::Server_Handle_GAMECMD(NetworkConnection& connection, NetworkPacket
...
             Server_Send_SHOWERROR(connection, STR_CANT_DO_THIS, STR_NETWORK_ACTION_RATE_LIMIT_MESSAGE);
             return;
         }
     }
     // Don't let clients send pause or quit
     else if (commandCommand == GAME_COMMAND_TOGGLE_PAUSE ||
-        commandCommand == GAME_COMMAND_LOAD_OR_QUIT
+             commandCommand == GAME_COMMAND_LOAD_OR_QUIT
```
Comment (janisozaur) [inline src/openrct2/network/network.cpp]: And here

**Comment by janisozaur** [comment]
`Network::Server_Send_TICK` sends srand0, srand1 and sprites' checksum which are collected before performing any of the game commands. In desync detection code, there is a variable `game_commands_processed_this_tick` which will only perform the check when there were no actions performed, to make sure the values match. With this change in place, could you refactor the desync check to only happen at start of tick and remove this said variable?

**Comment by janisozaur** [comment]
@ZehM4tt I looked at these changes and I have to take my last comment back. I see trying to squeeze the desync check in command processing would only cause even more misses, as it would only be checked for when commands are being processed. Instead, I applied a different patch: in command processing you return from the processing function on first entry from the future tick. I changed that `return` to `break` instead, so we still have a chance of finding a desync.

If you're fine with that, I'll merge it in.

**Comment by ZehM4tt** [comment]
You are right, almost missed that.

**Pull request merged**